tests/emscripten_runner.js

```diff
@@ -1,109 +1,115 @@
-      tty.output = [];
-    }
-  };
+      stream.write(Buffer.from(tty.output))
+      tty.output = []
+    },
+  }
 }
 
-function setupStreams(FS, TTY){
-  let mytty = FS.makedev(FS.createDevice.major++, 0);
-  let myttyerr = FS.makedev(FS.createDevice.major++, 0);
+function setupStreams(FS, TTY) {
+  let mytty = FS.makedev(FS.createDevice.major++, 0)
+  let myttyerr = FS.makedev(FS.createDevice.major++, 0)
   TTY.register(mytty, make_tty_ops(process.stdout))
   TTY.register(myttyerr, make_tty_ops(process.stderr))
-  FS.mkdev('/dev/mytty', mytty);
-  FS.mkdev('/dev/myttyerr', myttyerr);
-  FS.unlink('/dev/stdin');
-  FS.unlink('/dev/stdout');
-  FS.unlink('/dev/stderr');
-  FS.symlink('/dev/mytty', '/dev/stdin');
-  FS.symlink('/dev/mytty', '/dev/stdout');
-  FS.symlink('/dev/myttyerr', '/dev/stderr');
-  FS.closeStream(0);
-  FS.closeStream(1);
-  FS.closeStream(2);
-  var stdin = FS.open('/dev/stdin', 0);
-  var stdout = FS.open('/dev/stdout', 1);
-  var stderr = FS.open('/dev/stderr', 1);
+  FS.mkdev('/dev/mytty', mytty)
+  FS.mkdev('/dev/myttyerr', myttyerr)
+  FS.unlink('/dev/stdin')
+  FS.unlink('/dev/stdout')
+  FS.unlink('/dev/stderr')
+  FS.symlink('/dev/mytty', '/dev/stdin')
+  FS.symlink('/dev/mytty', '/dev/stdout')
+  FS.symlink('/dev/myttyerr', '/dev/stderr')
+  FS.closeStream(0)
+  FS.closeStream(1)
+  FS.closeStream(2)
+  FS.open('/dev/stdin', 0)
+  FS.open('/dev/stdout', 1)
+  FS.open('/dev/stderr', 1)
 }
 
-
 async function main() {
-  const wheelName = await findWheel(distDir);
-  const wheelURL = `file:${distDir}/${wheelName}`;
-  let errcode = 0;
+  const root_dir = path.resolve(__dirname, '..')
+  const wheel_path = await find_wheel(path.join(root_dir, 'dist'))
+  const wheel_url = `file:${wheel_path}`
+  let errcode = 0
   try {
-    pyodide = await loadPyodide();
-    const FS = pyodide.FS;
-    setupStreams(FS, pyodide._module.TTY);
-    const NODEFS = FS.filesystems.NODEFS;
-    FS.mkdir('/test_dir');
-    FS.mount(NODEFS, { root: testDir }, '/test_dir');
-    await pyodide.loadPackage(['micropip', 'pytest', 'pytz']);
-    const micropip = pyodide.pyimport('micropip');
-    await micropip.install('dirty-equals');
-    await micropip.install('hypothesis');
-    await micropip.install('pytest-speed');
-    await micropip.install(wheelURL);
-    const pytest = pyodide.pyimport('pytest');
-    FS.chdir("/test_dir");
-    errcode = pytest.main();
+    const pyodide = await loadPyodide()
+    const FS = pyodide.FS
+    setupStreams(FS, pyodide._module.TTY)
+    FS.mkdir('/test_dir')
+    FS.mount(FS.filesystems.NODEFS, {root: path.join(root_dir, 'tests')}, '/test_dir')
+    FS.chdir('/test_dir')
+    await pyodide.loadPackage(['micropip', 'pytest', 'pytz'])
+    // language=python
+    errcode = await pyodide.runPythonAsync(`
+import micropip
+
+await micropip.install('dirty-equals')
+await micropip.install('hypothesis')
+await micropip.install('pytest-speed')
+await micropip.install('${wheel_url}')
```

Comment:
This will be faster if you pass a single list to `micropip.install` because then it will make the http requests in parallel rather than sequentially.